chroma-bundles/include/find_deps.py

```diff
@@ -61,33 +62,40 @@ def req2pkgs(self, req):
         self.__req2pkgs[req] = providers
         return providers
 
-    def _find_deps(self, pkg, excludes):
+    def _find_deps(self, pkg, excludes, depth=0):
         # not interested in installed packages, just what's in the repo
         if pkg.__class__.__name__ == "RPMInstalledPackage":
             return
         if pkg in self.all_deps:
             return
+        print >>sys.stderr, "%spackage %s:" % (' ' * depth, pkg)
+        depth += 1
         self.all_deps.append(pkg)
         for tup in pkg.returnPrco('requires'):
             name = yum.misc.prco_tuple_to_string(tup)
             if name.startswith("rpmlib("):
                 continue
-            pkgs = [p for p in self.req2pkgs(name)
+            pkgs = [p for p in self.req2pkgs(name, depth - 1)
                     if not p.__class__.__name__ == "RPMInstalledPackage"]
+            #print "pkgs = %s" % pkgs
             if pkgs:
                 for npkg in pkgs:
                     if npkg.name in excludes:
                         continue
                     if npkg.name in self.all_pkg_names:
-                        self._find_deps(npkg, excludes)
+                        self._find_deps(npkg, excludes, depth)
                     else:
                         if npkg.name not in self.missing_deps:
+                            print >>sys.stderr, "adding to missing_deps 1"
                             self.missing_deps.append(npkg.name)
             else:
                 if name.find(' ') > -1:
                     name = name[0:name.find(' ')]
-                self.missing_deps.append(name)
+                if name not in self.all_pkg_names:
+                    print >>sys.stderr, "adding to missing_deps 2"
+                    self.missing_deps.append(name)
 
+        #print "missing_deps: %s:" % self.missing_deps
```

Comment:
It's frequently needed when debugging dependencies and I was getting tired of having to keep adding it.